internal/node/data_decoder_test.go

```diff
@@ -0,0 +1,29 @@
+package client
+
+import (
+	"encoding/base64"
+	"testing"
+)
+
+func TestTryDecodeBase64OrElse(t *testing.T) {
+	invalidBase64String := "not base64 encoded"
+	t.Log("Given invalid base64 input string")
+	{
+		t.Logf("\tTest: invalid base64 string: When string is invalid base64 encoded string `%s`", invalidBase64String)
+		result := TryDecodeBase64OrElse(invalidBase64String)
+		if string(result) == invalidBase64String {
+			t.Logf("\t\tResult `%s` should be equal to the original invalid base64 string `%s`", result, invalidBase64String)
+		}
+	}
+
+	str := "to be base64 encoded"
+	base64EncodedStr := base64.StdEncoding.EncodeToString([]byte(str))
+	t.Log("Given valid base64 input string")
+	{
+		t.Logf("\tTest: base64 string: When string is valid base64 encoded string: `%s`", base64EncodedStr)
+		result := TryDecodeBase64OrElse(base64EncodedStr)
+		if string(result) == str {
```

Comment:
also here i think assertion should be made to fail the test if values are not equal